src/main/ruby/truffleruby/core/kernel.rb

```diff
@@ -181,13 +181,8 @@ def `(str) #`
   end
   module_function :` # `
 
-  def =~(other)
-    warn "deprecated Object#=~ is called on #{Primitive.class(self)}; it always returns nil", uplevel: 1 if $VERBOSE
-    nil
-  end
-
   def !~(other)
-    r = self =~ other ? false : true
+    r = nil ? false : true
```

Comment:
`=~` returned `nil` as the default implementation. But, if a class defines `=~` then `!~` will call that version. You can see the dynamic call in CRuby:

https://github.com/ruby/ruby/blob/7b93e65e9f9cb7abfc7e19aff4da9d6e4c32de4b/object.c#L1629-L1634

As an example:

```ruby
class X
  def =~(other)
    puts :hi
  end
end

X.new !~ "abc"
```

That will print out 'hi' because `!~` delegates to `=~`.